apt_select/utils.py

```diff
@@ -22,11 +22,13 @@ class URLGetTextError(Exception):
 def get_text(url):
     """Return text from GET request response content"""
     try:
-        text = requests.get(url, headers=DEFAULT_REQUEST_HEADERS).text
+        result = requests.get(url, headers=DEFAULT_REQUEST_HEADERS)
+        if result.status_code != 200:
+            raise URLGetTextError("Failed to fetch mirror list: status code %d" % (result.status_code))
```

Comment:
This method is used to get html for launchpad mirror detail pages where this error text wouldn't apply if the response status is not 200. I think there error message here should be enough: https://github.com/jblakeman/apt-select/blob/d6daea219ebf1320d795cab2043741a2f278baf0/apt_select/mirrors.py#L77-L80

I'd prefer to leave it there.

To fix the bug though, I think I'd prefer to just use [`result.raise_for_status()`](https://2.python-requests.org/en/master/_modules/requests/models/#Response.raise_for_status) (which will raise an `requests.HTTPError`). Because at the moment, `apt-select` doesn't care about getting the text of any response which is not a successful one.

Reply:
Right, good idea! I pushed a fixup commit now. It now looks like this in case of failure:
```
WARNING: no country code provided. defaulting to US.
Getting list of mirrors...done.
Testing latency to mirror(s)
[79/80] 98%	connection to mirrors.ocf.berkeley.edu: timed out
[80/80] 100%
Getting list of launchpad URLs...https://launchpad.net/ubuntu/+archivemirros: 404 Client Error: Not Found for url: https://launchpad.net/ubuntu/+archivemirros
Unable to retrieve list of launchpad sites
Reverting to latency only1. la-mirrors.evowise.com: 1.28 ms
Selecting mirror http://la-mirrors.evowise.com/ubuntu/ ...
New config file saved to /home/knatten/code/apt-select/apt_select/sources.list
```

If you like this approach, I'll squash and update the commit message.

Reply:
@knatten Thanks! Hmm, the output in your above comment makes me realize that there needs to be a newline character after `Reverting to latency only` in the message. Sorry, not part of the original scope but would be great to fix as part of this.

Reply:
Sure, no problem. I added a commit for that now.

I also squashed the fixup and updated the first commit message with the changed behaviour.